forc-plugins/forc-debug/src/debugger/commands.rs

```diff
@@ -0,0 +1,419 @@
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum AbiMapping {
+    /// Local development ABI (no specific contract ID)
+    Local { abi_path: String },
+    /// Contract-specific ABI mapping
+    Contract {
+        contract_id: ContractId,
+        abi_path: String,
+    },
+}
+
+/// Response types for debug commands
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum DebugResponse {
+    /// Transaction started or continued with execution result
+    RunResult {
+        receipts: Vec<Receipt>,
+        breakpoint: Option<BreakpointHit>,
+    },
+    /// Command completed successfully with no data
+    Success,
+    /// Register values
+    Registers(Vec<RegisterValue>),
+    /// Memory contents
+    Memory(Vec<u8>),
+    /// Error occurred
+    Error(String),
+}
+
+/// Information about a breakpoint hit
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BreakpointHit {
+    pub contract: ContractId,
+    pub pc: u64,
+}
+
+/// Register value with metadata
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RegisterValue {
+    pub index: u32,
+    pub value: u64,
+    pub name: String,
+}
+
+impl DebugCommand {
+    /// Parse a command from CLI arguments
+    pub fn from_cli_args(args: &[String]) -> Result<Self, ArgumentError> {
+        if args.is_empty() {
+            return Err(ArgumentError::NotEnough {
+                expected: 1,
+                got: 0,
+            });
+        }
+
+        let cmd = &args[0];
+        let args = &args[1..];
+
+        match cmd.as_str() {
+            "start_tx" | "n" | "tx" | "new_tx" => {
+                Self::parse_start_tx(args).map_err(ArgumentError::Invalid)
+            }
+            "reset" => {
+                if !args.is_empty() {
+                    return Err(ArgumentError::Invalid(
+                        "reset command takes no arguments".to_string(),
+                    ));
+                }
+                Ok(DebugCommand::Reset)
+            }
+            "continue" | "c" => {
+                if !args.is_empty() {
+                    return Err(ArgumentError::Invalid(
+                        "continue command takes no arguments".to_string(),
+                    ));
+                }
+                Ok(DebugCommand::Continue)
+            }
+            "step" | "s" => Self::parse_step(args).map_err(ArgumentError::Invalid),
+            "breakpoint" | "bp" | "b" => {
+                Self::parse_breakpoint(args).map_err(ArgumentError::Invalid)
+            }
+            "register" | "r" | "reg" | "registers" => {
+                Self::parse_registers(args).map_err(ArgumentError::Invalid)
+            }
+            "memory" | "m" | "mem" => Self::parse_memory(args).map_err(ArgumentError::Invalid),
+            "quit" | "q" | "exit" => Ok(DebugCommand::Quit),
+            _ => Err(ArgumentError::UnknownCommand(cmd.to_string())),
+        }
+    }
+
+    fn parse_start_tx(args: &[String]) -> Result<Self, String> {
```

Comment:
maybe we should bring over the doc comment explaining the usage that was removed and place it here? 

```
/// Parse a start_tx command from CLI arguments
///
/// Handles two distinct modes of operation:
/// 1. Local Development: `tx transaction.json abi.json`
/// 2. Contract-specific: `tx transaction.json --abi <contract_id>:<abi_file.json>`
```

Reply:
https://github.com/FuelLabs/sway/pull/7320/commits/b9f50545fcbe3e2bf009b5f8cef9c60ff0802b25